### PR TITLE
Fix: reload state on window refresh

### PR DESCRIPTION
**Pull request by neonnoon**
### Description

Properly restore the panel after a window refresh. Make sure that event handlers are registered.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated

### CODE REVIEW

File: src/snyk/common/views/webviewProvider.ts [hunk 1]
```diff
@@ -25,9 +25,12 @@ export abstract class WebviewProvider<ViewModel> implements IWebViewProvider<Vie
   restorePanel(panel: vscode.WebviewPanel): void {
     if (this.panel) this.panel.dispose();
     this.panel = panel;
+    this.registerListeners();
+    this.panel.webview.html = this.getHtmlForWebview(this.panel.webview);
```
Comment (michelkaporin) [inline src/snyk/common/views/webviewProvider.ts:29]: There's a need to call some custom logic in [codeSuggestionWebviewProvider.ts](https://github.com/snyk/vscode-extension/pull/248/files#diff-3f69198f2ef12d4c2a39fa3d622621d106fe54db326e22a7d8869a319845760b). It is setting `this.suggestion` on the provider, which doesn't get set from here.

We need it for correct functioning of `checkCurrentSuggestion()` that checks if suggestion panel contains an existing issue from the run analysis.

Is there a way still to reuse `showPanel()` to keep custom provider logic in place?

**Review by michelkaporin — APPROVED** [review]

File: src/snyk/common/views/webviewProvider.ts (hunk 1 above)
Comment (neonnoon) [reply src/snyk/common/views/webviewProvider.ts:29]: Hmm, I see.. We might want to split showPanel and updatePanel or something of the sort.
I will have a closer look and make a suggestion.

Comment (michelkaporin) [reply src/snyk/common/views/webviewProvider.ts:29]: You're probably right. You can ditch `checkCurrentSuggestion()` function altogether then. OSS provider doesn't have this check, thus we're consistent.

Comment (neonnoon) [reply src/snyk/common/views/webviewProvider.ts:29]: So, the implementation in the PR is consistent. We don't set `this.suggestion` and the check [here](https://github.com/snyk/vscode-extension/blob/main/src/snyk/snykCode/views/suggestion/codeSuggestionWebviewProvider.ts#L70) will just skip it.

Comment (michelkaporin) [reply src/snyk/common/views/webviewProvider.ts:29]: Yep, let's get this out. 🚀

**Review by michelkaporin — APPROVED** [review]
Great work @neonnoon! Appreciate you tackling this one 💪

**Pull request merged**